### PR TITLE
Fixing boot menu issues

### PR DESCRIPTION
**Pull request by tpw56j**
- fix menu item generation to return to parent menu (pxe/ipxe)
- the large boot menu took a very long time to generate due to the template being read from disk in a loop for each menu item at each menu level
- in a large boot menu it is difficult to navigate by the short names of profiles, images and systems, so I suggest adding a display_name property for them to be displayed in the boot menu

### CODE REVIEW

**Comment by tpw56j** [comment]
@SchoolGuy I've noticed that cobbler no longer generates a boot menu for Images because the corresponding code has been removed from write_pxe_file. If this is correct, then get_images_menu also becomes unnecessary and should be removed?

**Comment by SchoolGuy** [comment]
@tpw56j Oh god no. This is a bug that is currently open (see #3094). I do want to have those entries. Did I remove that? And if yes in which commit?

File: cobbler/items/menu.py [hunk 1]
```diff
@@ -107,24 +107,7 @@ def children(self, value: List[str]):
...
-            for name in value:
-                menu = self.api.find_menu(name=name)
-                if menu is not None:
-                    self._children.append(name)
-                else:
-                    self.logger.warning(
-                        'Menu with the name "%s" did not exist. Skipping setting as a child!',
-                        name,
-                    )
+        self._children = value
```
Comment (SchoolGuy) [inline cobbler/items/menu.py:110]: Why did you remove this code? I added it in the past so you can pass a list of names or objects. If this is correct, I would love if we could put the reasoning in a commit message.

File: cobbler/tftpgen.py [hunk 2]
```diff
@@ -20,6 +20,7 @@
 from cobbler.utils import input_converters
 from cobbler.validate import validate_autoinstall_script_name
 from cobbler.utils import filesystem_helpers
+from cobbler.items.menu import Menu
```
Comment (SchoolGuy) [inline cobbler/tftpgen.py:23]: See comment below.

```suggestion
```

File: cobbler/tftpgen.py [hunk 3]
```diff
@@ -456,29 +457,30 @@ def get_submenus(self, menu, metadata: dict, arch: enums.Archs):
...
+                    )
+                    menu_labels[boot_loader].append(
+                        {"name": child.name, "display_name": display_name}
+                    )
 
         for boot_loader in boot_loaders:
             if (
                 boot_loader in nested_menu_items
-                and nested_menu_items[boot_loader] != ""
+                and nested_menu_items[boot_loader][-2:] == "\n\n"
```
Comment (SchoolGuy) [inline cobbler/tftpgen.py:476]: Why should the last two characters be newlines? I would love a comment here.

File: cobbler/tftpgen.py [hunk 4]
```diff
@@ -531,19 +533,22 @@ def get_profiles_menu(self, menu, metadata, arch: enums.Archs):
...
                         menu_labels["ipxe"].append(
-                            {"name": profile.name, "display_name": profile.name}
+                            {"name": profile.name, "display_name": display_name}
                         )
 
         metadata["menu_items"] = current_menu_items
         metadata["menu_labels"] = menu_labels
 
-    def get_images_menu(self, menu, metadata, arch: enums.Archs):
+    def get_images_menu(self, menu: Menu, metadata: dict, arch: enums.Archs):
```
Comment (SchoolGuy) [inline cobbler/tftpgen.py:551]: See comment above for cyclic imports

```suggestion
    def get_images_menu(self, menu, metadata: dict, arch: enums.Archs):
```

**Comment by tpw56j** [comment]
> @tpw56j Oh god no. This is a bug that is currently open (see #3094). I do want to have those entries. Did I remove that? And if yes in which commit?

@SchoolGuy I was wrong, this code was not removed. In fact, the problem is in the mismatch of data types. I've added the fix to the new commit.

File: cobbler/items/menu.py (hunk 1 above)
Comment (tpw56j) [reply cobbler/items/menu.py:110]: The only reason I removed all the code except for one line in children.setter was that in profile.py this code also consists of exactly one line. Initially, I wrote the menu.py code and now I decided to check whether the implementation of properties in it corresponds to the same properties in other classes. I thought that the code in profile.py was more correct, but in menu.py it was simply forgotten to fix it. The children property is used in tftpgen.py when forming submenus and it is important that it is implemented correctly. I haven't examined the item.get_children code, so I can't say which code is more correct. Revert the old code back?

Comment (SchoolGuy) [reply cobbler/items/menu.py:110]: Since we have not type guarantees (after all we only add type hints to the code) and the old code contains the functionality the new code has, I would indeed vote for reverting it to the previous state. If you think you are absolutely certain that this code is not required then I gladly accept the changes your have proposed.

File: cobbler/tftpgen.py [hunk 5]
```diff
@@ -423,7 +424,7 @@ def get_menu_items(self, arch: Optional[enums.Archs] = None) -> dict:
         """
         return self.get_menu_level(None, arch)
 
-    def get_submenus(self, menu, metadata: dict, arch: enums.Archs):
+    def get_submenus(self, menu: Menu, metadata: dict, arch: enums.Archs):
```
[the remark this answers is not in the document]
Comment (SchoolGuy) [reply cobbler/tftpgen.py:427]: Just as a secondary note. I am aware that with Python 3.10 this just works out of the box, however many places where this will be run are still on Python 3.6 and there we might break things.

File: cobbler/items/menu.py (hunk 1 above)
Comment (tpw56j) [reply cobbler/items/menu.py]: I agree, I returned this code back.

File: cobbler/tftpgen.py (hunk 3 above)
Comment (tpw56j) [reply cobbler/tftpgen.py]: This code turned out to be superfluous, I removed it.

**Comment by SchoolGuy** [comment]
@tpw56j If this PR is considered ready by you, I would love if you could fix the Black linter.

**Comment by tpw56j** [comment]
@SchoolGuy PR is ready for a new review.
Additional changes in new commits:
- menu_name -> menu_label in grub.pxe/ipxe templates. At the time of installation, apart from the boot menu, there is nothing else on the screen at hand and it is difficult to remember what is hidden under the abbreviation of the short name in {menu|profile|image|system}.name.
- for ipxe added submenu indicator as it looks for grub. Unlike grub, the indicator cannot be added to the template and had to be added to the code.
- added fix for #3094

**Review by SchoolGuy — APPROVED** [review]
All my questions were answered and the code is LGTM from my point of view.

**Comment by SchoolGuy** [comment]
@tpw56j I just saw that your commits are not signed. If possible it would be cool if you could do that. If not I will still merge.

**Comment by tpw56j** [comment]
@SchoolGuy I signed the commits.
